tools/model-registry/tests/test_smoke.py

```
@@ -92,12 +94,62 @@ def post(url, headers, payload):
 
-def test_network_error_on_a_configured_link_fails(tmp_path):
+def test_permission_denied_is_a_dead_id_fail(tmp_path):
+    # A 403 permission denial (the key exists but cannot serve this id) is the
+    # #298 access failure — a registry defect — not an account-funding excuse.
     def post(url, headers, payload):
-        raise OSError("connection refused")
+        return 403, '{"error":{"type":"permission_error","message":"no access to model"}}'
     lines, code = smoke.smoke_chain(load(tmp_path), "review", {"ZAI_KEY": "zk"}, post)
     assert code == 1
-    assert any("connection refused" in l for l in lines)
+    assert any(l.startswith("FAIL") for l in lines)
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)